_rules/SC2-4-6-descriptive-labels.md

```diff
@@ -148,23 +199,58 @@ Label is included in accessibility tree, but not visible, and does not describe
 <input aria-labelledby="label_fname" type="text" name="fname"/>
 ```
 
+#### Failed Example 6
+
+Accessible name created through "aria-label" does not describes the purpose of the associated element.
+
+```html
+<input aria-label="Menu" id="fname" type="text" name="fname"/>
+```
+
+#### Failed Example 7
+
+The `label` in itself does not describe the purpose of the form field, and while the context can be relied upon to differentiate form fields from each other, it is not sufficient to rely on the context for describing the entire purpose of the form field.
+
+```html
+<h2>Name</h2>
+<label>Fill in: <input id="name" type="text" name="name"/></label>
+````
+
+#### Failed Example 8
+
+The descriptive label coded with the `label` element is through the accessible name calculation overwritten by an `aria-label` that is not descriptive.
+
+```html
+<label for="fname">First name:</label>
+<input id="fname" type="text" name="fname" aria-label="Fill in:"/>
+```
+
+#### Failed Example 9
+
+The non-descriptive label coded with the `label` element is through the accessible name calculation overwritten by an `aria-label` that is descriptive. However, the visible label is still not descriptive.
+
+```html
+<label for="fname">Fill in:</label>
+<input id="fname" type="text" name="fname" aria-label="First name:"/>
+```
+
 ### Inapplicable
 
-#### Inapplicable example 1
+#### Inapplicable Example 1
 
 `Label` that is neither visible to users, nor included in the accessibility tree.
 
 ```html
-<label for="fname" style:"display:none;">First name:</label>
+<label for="fname" style="display:none;">First name:</label>
 <input id="fname" type="text" name="fname"/>
 ```
 
-#### Inapplicable example 2
+#### Inapplicable Example 2
 
 Programatically associated `p` element that is neither visible nor included in the accessibility tree.
```

Comment:
On Chrome the `input` element has an accessible name given by the p#label_fname so this should be applicable

Reply:
Yes, aria-* relationship are not affected by the visibility of the element (display:none, visibility:hidden have no impact on the calculated label).